### ISSUE TITLE
Compiling the .json files.

### ISSUE THREAD

**Issue by ArtoriaPendragon0** [UlyssesWu/FreeMote#82]
I decompiled a file named scenario_info.psb.m in order to translate a game. A scenario_info.psb.m.json was generated with a folder named scenario containing many .txt.scn files. I decompiled the .txt.scn files, edited the text then used PsBuild to have txt.scn files again then I replaced them with the original ones in the scenario folder. Then, I used PsBuild on the scenario_info.psb.m.json file and when I put the new .psb file in the game, there are no changes in the game and when I decompiled the .psb file again I didn't find the changes in the new scenario folder either. I'm assuming that I'm doing something wrong when trying to compile. I tried everything possible but I'm stuck. What am I doing wrong?

**Comment by UlyssesWu** [comment]
You should provide some screenshots about your "scenario" folder. But for the first guess, you may have left the unmodified "xxx.txt.scn.m.json" file in "scenario" folder (you can also call it as project folder or source folder).

If you read the instructions in `PsBuild info-psb -h`, it tells you:

>  If there are both `a.psb.m` and `a.psb.m.json` in the source folder, `.json` will be used (unless using `--packed`).

So, if your `xxx.txt.scn.m.json` file is not modified and not deleted, and you just put a modified `xxx.txt.scn.m` file in it, nothing will change. 
You should either leave a modified `xxx.txt.scn.m.json` file (and its resource folder and resx.json if exists) in "scenario" folder, and delete the `xxx.txt.scn.m` file, then PsBuild will compile `xxx.txt.scn.m.json` to `xxx.txt.scn.m` when building;
Or you should make sure there is a modified `xxx.txt.scn.m` file in "scenario" folder and there is no json files about it (since json is preferred than PSB for the same name), then PsBuild will just use the `xxx.txt.scn.m` when building. 

It's just intuitive isn't it?

**Comment by ArtoriaPendragon0** [comment]
No, I made sure to not leave the unmodified .json file. I just replaced the modifed .txt.scn file with the unmodified one and deleted the .json (the modified and the unmodified one). I even deleted the .txt.scn I wanted to modify from the scenario folder and then used PsBuild but the text present in the .txt.scn I deleted appeared in the game. I uploaded the screenshot of the scenario folder and I uploaded a screenshot of the folder in which info_scenario.psb.m.json file is present (it's not in the scenario folder, although it's the same result even when I put it in the scenario folder). 
![2022-06-09](https://user-images.githubusercontent.com/106391503/172794735-666ae1ab-a2eb-459d-ab3b-13262fe2af57.png)
![Capture](https://user-images.githubusercontent.com/106391503/172795209-2280386d-675b-4374-92e0-979fd9dc8c34.PNG)

**Comment by UlyssesWu** [comment]
It seems that you didn't use `PsBuild info-psb` command to build the PSB. You just compiled a scenario_info.psb.m (use `PsBuild scenario_info.psb.m.json`), but the scenario_body.bin is not modified (see Date modified).

**Comment by ArtoriaPendragon0** [comment]
I did this from the very start but this message popped up: "Json is not an ArchiveInfo PSB."

**Comment by UlyssesWu** [comment]
Attach screenshots (include your input command and program output), and attach the `scenario_info.psb.m.json` and `scenario_info.psb.m.resx.json` file.

**Comment by ArtoriaPendragon0** [comment]
![2022-06-09 (3)](https://user-images.githubusercontent.com/106391503/172825088-8e00c9ba-20c8-45d9-8324-91f0c0f16549.png)
[FreeMoteToolkit.zip](https://github.com/UlyssesWu/FreeMote/files/8869431/FreeMoteToolkit.zip)

**Comment by UlyssesWu** [comment]
OK, this is a special PSB, not identified by FreeMote as an "ArchiveInfo" type. I will try to fix it soon.

**Comment by ArtoriaPendragon0** [comment]
> OK, this is a special PSB, not identified by FreeMote as an "ArchiveInfo" type. I will try to fix it soon.

Thank you very much!

**Comment by UlyssesWu** [comment]
Try again with this dev version: https://ci.appveyor.com/project/UlyssesWu/freemote/build/artifacts

You have to unpack with this version first, do not use the old jsons. But you can use the modified .txt.scn file.

**Comment by ArtoriaPendragon0** [comment]
I tried with the freemote you uploaded here. That's what I got:
![2022-06-09 (4)](https://user-images.githubusercontent.com/106391503/172834276-c4e014d8-418b-43d9-8b06-21293475a57e.png)

**Comment by UlyssesWu** [comment]
Try again: [ci.appveyor.com/project/UlyssesWu/freemote/build/artifacts](https://ci.appveyor.com/project/UlyssesWu/freemote/build/artifacts) (same url)

Just try `PsBuild info-psb`, no need to extract again this time.

**Comment by ArtoriaPendragon0** [comment]
Worked like a charm! Thank you very much I really appreciate it
